### PR TITLE
I77 better xml validation errors

### PR DESCRIPTION
**Pull request by ZoningFraction**

### CODE REVIEW

File: scripts/dialog_xml2json.py [hunk 1]
```diff
@@ -863,24 +870,29 @@ def main(argv):
 
     # load dialogue from XML
     if hasattr(config, 'common_dialog_main'):
```
Comment (Tereza-Pytelova) [inline scripts/dialog_xml2json.py]: You can use `getOptionalParameter` function.

File: ci/unit_tests/dialog_xml2json/main_data/dialog_schema.xml [hunk 2]
```diff
@@ -0,0 +1,379 @@
+<?xml version="1.0"?>
```
Comment (Tereza-Pytelova) [inline ci/unit_tests/dialog_xml2json/main_data/dialog_schema.xml]: Please use `dialogSchemaPath` from `BaseTestCaseCapture` class

File: ci/unit_tests/dialog_xml2json/main_test.py [hunk 3]
```diff
@@ -51,3 +53,38 @@ def test_mainValidActions(self):
...
+
+
+    def test_validation(self):
+        """Tests if the script runs successfully with valid dialogs and fails with invalid dialog files."""
+
+        schemaPath = os.path.join(self.dataBasePath, "dialog_schema.xml")
+        validInputsFolder = os.path.join(self.dataBasePath, "validInputsAccordingToSchema")
+        invalidInputsFolder = os.path.join(self.dataBasePath, "invalidInputsAccordingToSchema")
+
+        for f in os.listdir(validInputsFolder):
```
Comment (Tereza-Pytelova) [inline ci/unit_tests/dialog_xml2json/main_test.py]: I'm still not sure if this is better than call each file separately.. You can then customize log messages.. And avoid weird filenames (goto_missingTarget.xml  when target present etc.)

File: scripts/dialog_xml2json.py [hunk 4]
```diff
@@ -189,14 +192,18 @@ def importNodes(root, config):
         defaultNode = root[len(root)-1]
 
     for node in root.findall('import'):
-        logger.verbose('Importing %s', os.path.join(os.path.dirname(getattr(config, 'common_dialog_main')),node.text))
-        importPath = node.text.split('/')
-        importTree = LET.parse(os.path.join(os.path.dirname(getattr(config, 'common_dialog_main')),*importPath))
+        logger.verbose('Importing %s', os.path.join(os.path.dirname(getattr(config, 'common_dialog_main')), node.text))
```
Comment (Tereza-Pytelova) [inline scripts/dialog_xml2json.py:198]: I would use `mainDialogPath = getRequiredParameter(config, 'common_dialog_main')`

File: scripts/dialog_xml2json.py [hunk 5]
```diff
@@ -863,24 +870,29 @@ def main(argv):
...
     # load dialogue from XML
     if hasattr(config, 'common_dialog_main'):
-        #TODO might need UTF-8
-        dialogTree = LET.parse(getattr(config, 'common_dialog_main'))
+        dialogTreeFile = getattr(config, 'common_dialog_main')
     else:
-        dialogTree = LET.parse(sys.stdin)
+        dialogTreeFile = sys.stdin
+
+    if not os.path.exists(dialogTreeFile):
```
Comment (Tereza-Pytelova) [inline scripts/dialog_xml2json.py:877]: Does this work even for the `sys.stdin`?

**Review by Tereza-Pytelova — CHANGES_REQUESTED** [review]
Please consider adding changes from comments

File: ci/unit_tests/dialog_xml2json/main_test.py (hunk 3 above)
Comment (ZoningFraction) [reply ci/unit_tests/dialog_xml2json/main_test.py]: You are right, will fix this.

File: scripts/dialog_xml2json.py (hunk 5 above)
Comment (Tereza-Pytelova) [reply scripts/dialog_xml2json.py:877]: more in #365

**Review by Tereza-Pytelova — APPROVED** [review]

**Pull request merged**